src/main/java/seedu/address/model/book/BookName.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.model.book;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Book's name in LibTask.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBookName(String)}
+ */
+public class BookName {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Book names should only contain alphanumeric characters and spaces, "
+        + "and it should not be blank, and it must start with an alphanumeric character.";
+
+    /*
+     * The first character of the name must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     * The name should also contain at least one alphanumeric character,
+     * and contains no non-alphanumeric characters other than ' character and : character.
+     */
```

Comment:
The message and javadocs need to be changed as well to reflect the reflections as explained below on the restrictions for the book names.